tests/lib/newlib/newlib_api/src/main.c

```diff
@@ -0,0 +1,284 @@
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <ztest_error_hook.h>
+#include <toolchain.h>
+#include <stdio.h>
+#include <stddef.h>
+
+extern int _gettimeofday(struct timeval *__tp, void *__tzp);
+extern int *__errno(void);
+extern void __chk_fail(void);
+extern void *_sbrk(intptr_t count);
+extern int _read(int fd, char *buf, int nbytes);
+extern int _write(int fd, const void *buf, int nbytes);
+extern int _open(const char *name, int mode);
+extern int _close(int file);
+extern int _lseek(int file, int ptr, int dir);
+extern int _isatty(int file);
+extern int _kill(int i, int j);
+extern int _getpid(void);
+extern int _fstat(int file, struct stat *st);
+extern void _exit(int status);
+extern void __stdin_hook_install(unsigned char (*hook)(void));
+extern int _stdout_hook_default(int c);
+
+#define TEST_INPUT 1
+
+/**
+ * @brief Test newlib apis without CONFIG_POSIX_API configuration
+ *
+ * @details Verify the apis which are implemented in libc/newlib/.
+ * And check it could be invoked which is default implemented with
+ * different input.
+ *
+ * @see _gettimeofday(), __errno(), _isatty(), _kill()
+ * _getpid(), _fstat()
+ */
+void test_newlib_api(void)
+{
+#ifndef CONFIG_POSIX_API
+	ztest_test_skip();
+#endif
+
+	struct timeval tp;
+	int *ptr;
+	int ret;
+	struct stat st;
+
+	ret = _gettimeofday(&tp, NULL);
+	zassert_equal(ret, 0, NULL);
+
+	ptr = __errno();
+	zassert_equal(*ptr, 0, NULL);
+
+	ret = _isatty(12);
+	zassert_false(ret, NULL);
+
+	ret = _isatty(-12);
+	zassert_true(ret, NULL);
+
+	ret = _isatty(0);
+	zassert_true(ret, NULL);
+
+	ret = _kill(123, 3);
+	zassert_equal(ret, 0, NULL);
+
+	ret = _kill(-123, 3);
+	zassert_equal(ret, 0, NULL);
+
+	ret = _kill(123, -3);
+	zassert_equal(ret, 0, NULL);
+
+	ret = _kill(-123, -3);
+	zassert_equal(ret, 0, NULL);
+
+	ret = _getpid();
+	zassert_equal(ret, 0, NULL);
+
+	ret = _fstat(13, &st);
+	zassert_equal(ret, 0, NULL);
+
+	ret = _fstat(-13, &st);
+	zassert_equal(ret, 0, NULL);
+
+	ztest_set_fault_valid(true);
+	ret = _fstat(-13, NULL);
+	zassert_equal(ret, 0, NULL);
+}
+
+/**
+ * @brief Test _sbrk API
+ *
+ * @details Invoke _sbrk without defined by self to check newlib
+ * api work or not. Give it with normal input and error input to
+ * verify this API.
+ *
+ * @see _sbrk().
+ */
+void test_sbrk_error(void)
+{
+	long count;
+	void *p;
+	int ret;
+
+	count = 1;
+	p = _sbrk(count);
+	ret = *(int *)p;
+	zassert_equal(ret, 0, NULL);
+
+	/*Use the count more than max heap size*/
+	ztest_set_fault_valid(true);
+	count = 0x1fffff;
+	p = _sbrk(count);
+}
+
+/**
+ * @brief Test __chk_fail API
+ *
+ * @details Invoke __chk_fail directly to check whether
+ * it will cause fatal error.
+ *
+ * @see __chk_fail()
+ */
+void test_chk_fail(void)
+{
+	/*__chk_fail will invoke z_opps(), should enable this framework*/
```

Comment:
```suggestion
	/* __chk_fail will invoke z_oops(), should enable this framework */
```

1. Typo
2. Add space after and before `/*` and `*/`, respectively.